### DISCUSSION TITLE
Properly using find_essential method

### DISCUSSION THREAD

**Discussion by ajuric** [kornia/kornia#2726] [Questions]
Hi!

I need help with properly using the [find_essential](https://kornia.readthedocs.io/en/latest/geometry.epipolar.html#kornia.geometry.epipolar.find_essential) method.

I have 2D point correspondences for image pair and I know that from that data I can estimate the fundamental matrix. If I have fundamental matrix, I can multiply it with calibration matrices to get the essential matrix.

In opencv, method for estimating the essential matrix takes also K (calibration) together with corresponding points, but in Kornia it only takes the points (no calibration needed?) and that makes me confused.

I have seen an example ([here](https://github.com/cvg/glue-factory/blob/main/gluefactory/robust_estimators/relative_pose/opencv.py#L33)) where points are normalized before estimating the essential matrix and than for K an eye matrix is given.

Should it be the same here in Kornia?

@ducha-aiki I'm tagging you since I have seen that you have LGTM-ed that merge request integrating this Nister's 5-point alg for the estimation. :)

So, if I have `kpts0`, `kpts1`, `K`, with shapes:
* `kpts0`: (1, N, 2)
* `kpts1`: (1, N, 2)
* `K`: (1, 3, 3)
what do I need to do to properly use the `find_essential` method in Kornia and get the essential matrix?

Thanks!

**Comment by ducha-aiki** [comment]
Hi,

>I have seen an example ([here](https://github.com/cvg/glue-factory/blob/main/gluefactory/robust_estimators/relative_pose/opencv.py#L33)) where points are normalized before estimating the essential matrix and than for K an eye matrix is given.
>Should it be the same here in Kornia?

Yes, exactly :)
